File: apps/ios/Roundtable/Sources/Models/Message.swift
import Foundation

enum MessageRole: String, Codable, Sendable {
    case system
    case assistant
    case user
    case moderator
}

struct Message: Codable, Identifiable, Sendable {
    let id: String
    let sessionId: String
    let participantId: String
    let participantName: String
    let role: MessageRole
    let content: String
    let turnNumber: Int
    var timestamp: String?
}

File: packages/shared/src/types/message.ts
export type MessageRole = "system" | "assistant" | "user" | "moderator";

export interface Message {
  id: string;
  sessionId: string;
  participantId: string;
  participantName: string;
  role: MessageRole;
  content: string;
  turnNumber: number;
  timestamp: string;
}

export interface TranscriptEntry extends Message {
  provider: string;
  model: string;
  tokenCount?: number;
  durationMs?: number;
}

export interface StreamingChunk {
  sessionId: string;
  participantId: string;
  content: string;
  done: boolean;
}
